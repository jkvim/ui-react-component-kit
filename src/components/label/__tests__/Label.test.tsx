import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Label } from '../';

describe('Label', () => {

  describe('props', () => {

    const wrapper = mount(
      <Label htmlFor="id" value="label" type="standard" hasError={true} />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `htmlFor` in props', () => {
      expect(wrapper.props().htmlFor).toBe('id');
    });

    it('should pass `value` in props', () => {
      expect(wrapper.props().value).toBe('label');
    });

    it('should pass `type` in props', () => {
      expect(wrapper.props().type).toBe('standard');
    });

    it('should pass `hasError` in props', () => {
      expect(wrapper.props().hasError).toBe(true);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<Label htmlFor="id" value="label" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot standard', () => {
    const wrapper = mount(<Label htmlFor="id" value="label" type="standard" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot radio', () => {
    const wrapper = mount(<Label htmlFor="id" value="label" type="radio" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot checkbox', () => {
    const wrapper = mount(<Label htmlFor="id" value="label" type="checkbox" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(<Label htmlFor="id" value="label" hasError={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});