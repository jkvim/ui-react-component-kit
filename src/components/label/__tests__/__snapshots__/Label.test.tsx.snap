// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label props should match snapshot 1`] = `
<Label
  hasError={true}
  htmlFor="id"
  type="standard"
  value="label"
>
  <styled.label
    hasError={true}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <label
      className="sc-bdVaJa esUHjj"
      htmlFor="id"
    >
      label
    </label>
  </styled.label>
</Label>
`;

exports[`Label snapshot basic 1`] = `
<Label
  htmlFor="id"
  value="label"
>
  <styled.label
    hasError={false}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <label
      className="sc-bdVaJa mlfRZ"
      htmlFor="id"
    >
      label
    </label>
  </styled.label>
</Label>
`;

exports[`Label snapshot checkbox 1`] = `
<Label
  htmlFor="id"
  type="checkbox"
  value="label"
>
  <styled.label
    hasError={false}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="checkbox"
  >
    <label
      className="sc-bwzfXH guwhKP"
      htmlFor="id"
      type="checkbox"
    >
      label
    </label>
  </styled.label>
</Label>
`;

exports[`Label snapshot error 1`] = `
<Label
  hasError={true}
  htmlFor="id"
  value="label"
>
  <styled.label
    hasError={true}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <label
      className="sc-bdVaJa esUHjj"
      htmlFor="id"
    >
      label
    </label>
  </styled.label>
</Label>
`;

exports[`Label snapshot radio 1`] = `
<Label
  htmlFor="id"
  type="radio"
  value="label"
>
  <styled.label
    hasError={false}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="radio"
  >
    <label
      className="sc-bwzfXH cuKRqK"
      htmlFor="id"
      type="radio"
    >
      label
    </label>
  </styled.label>
</Label>
`;

exports[`Label snapshot standard 1`] = `
<Label
  htmlFor="id"
  type="standard"
  value="label"
>
  <styled.label
    hasError={false}
    htmlFor="id"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <label
      className="sc-bdVaJa mlfRZ"
      htmlFor="id"
    >
      label
    </label>
  </styled.label>
</Label>
`;
